pkg/catalog/catalog.go

```diff
@@ -64,15 +65,24 @@ func getRepos(ctx context.Context, c *model.Catalog, ua string, skipTLSVerify bo
 		return nil, err
 	}
 
-	list := make([]*vcs.Repository, len(repos))
+	var nameReg *regexp.Regexp
+	if c.FilterPattern != "" {
+		nameReg = regexp.MustCompile(c.FilterPattern)
```

Comment:
Avoid using MustCompile, it causes panic

Reply:
updated